nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestEncodeContent.java

```diff
@@ -123,43 +122,275 @@ public void testFailDecodeNotBase32() throws IOException {
+        final String expectedOutput = "TG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQsIGNvbnNlY3RldHVyIGFkaXBpc2NpbmcgZWxpdCwg"
+            + "c2VkIGRvIGVpdXNtb2QgdGVtcG9yIGluY2lkaWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWdu"
+            + "YSBhbGlxdWEu";
+
+        // Setting a low value for `lineLength` but single line true ensures that `lineLength` is ignored
+        executeTestHelper(EncodingMode.ENCODE,
+            EncodingType.BASE64_ENCODING,
+            LOREM_IPSUM,
+            LineOutputMode.SINGLE_LINE, // set true to output single line
+            2,                                      // set a low value >= 0
+            System.lineSeparator(),
+            expectedOutput,
+            EncodeContent.REL_SUCCESS);
+    }
+
+    private void executeTestSuccessHelper(final DescribedValue mode,
+        final DescribedValue encodingType,
+        final String input,
+        final String expectedOutput) {
+        executeTestSuccessHelper(mode, encodingType, input, LineOutputMode.MULTIPLE_LINES, expectedOutput);
+    }
+
+    private void executeTestSuccessHelper(final DescribedValue mode,
+        final DescribedValue encodingType,
+        final String input,
+        final DescribedValue outputToSingleLine,
+        final String expectedOutput) {
+        executeTestHelper(mode, encodingType, input, outputToSingleLine, expectedOutput, EncodeContent.REL_SUCCESS);
+    }
+
+    private void executeTestHelper(final DescribedValue mode,
+        final DescribedValue encodingType,
+        final String input,
+        final DescribedValue outputToSingleLine,
+        final String expectedOutput,
+        final Relationship routedTo) {
+        executeTestHelper(mode,
+            encodingType,
+            input,
+            outputToSingleLine,
+            76,
+            System.lineSeparator(),
+            expectedOutput,
+            routedTo);
+    }
+
+    private void executeTestHelper(final DescribedValue mode,
+        final DescribedValue encodingType,
+        final String input,
+        final DescribedValue outputToSingleLine,
+        final Integer lineLength,
+        final String lineSeparator,
+        final String expectedOutput,
+        final Relationship routedTo) {
+
+        testRunner.setProperty(EncodeContent.MODE, mode.getValue());
+        testRunner.setProperty(EncodeContent.ENCODING, encodingType.getValue());
+        testRunner.setProperty(EncodeContent.LINE_OUTPUT_MODE, outputToSingleLine.getValue());
+        testRunner.setProperty(EncodeContent.ENCODED_LINE_LENGTH, Integer.toString(lineLength));
+        testRunner.setProperty(EncodeContent.ENCODED_LINE_SEPARATOR, lineSeparator);
+
+        testRunner.enqueue(input);
         testRunner.run();
 
-        testRunner.assertAllFlowFilesTransferred(EncodeContent.REL_FAILURE, 1);
+        final MockFlowFile result = testRunner.getFlowFilesForRelationship(routedTo).get(0);
+        assertEquals(expectedOutput, result.getContent());
```

Comment:
Intellij has the following issue 

> Call to 'assertEquals()' from 'Assert' should be replaced with a call to a method from 'org.junit.jupiter.api.Assertions'

Reply:
https://github.com/apache/nifi/pull/8417/commits/4e7732fccb2b70b8f44fb571c0d3d86e8320fb40